src/app/subject/store/sensors-data/sensors-data.reducer.ts

```diff
@@ -24,23 +24,22 @@ export const initialState: State = adapter.getInitialState({
   areLoaded: {},
   dates: [],
   tooltipDate: null,
-  timeFrame: { start: null, end: null },
+  timeFrame: { startDateTime: null, endDateTime: null },
   timeWindow: 'TEN_SECOND',
   descriptiveStatistic: DescriptiveStatistic.MEDIAN
 })
 
 export function reducer(state = initialState, action: actions.Actions): State {
   switch (action.type) {
     case actions.UPDATE_DATES: {
+      const end = new Date(state.timeFrame.endDateTime)
+      const start = new Date(state.timeFrame.startDateTime)
       const iterations =
-        (state.timeFrame.end - state.timeFrame.start) /
-        TimeWindow[state.timeWindow]
+        (end.getTime() - start.getTime()) / TimeWindow[state.timeWindow]
 
       const dates = []
       for (let i = 0; i < iterations; i++) {
-        dates[i] = new Date(
-          state.timeFrame.start + TimeWindow[state.timeWindow] * i
-        )
+        dates[i] = new Date(start.getTime() + TimeWindow[state.timeWindow] * i)
```

Comment:
Where do we use these `dates` ? The rest-api also gives the `timeStamps` of each data-record that is returned, which is effectively , what these lines of code do too. I'm a little confused why this is calculated from dashboard side too. Probably on a different application.

Reply:
Yeah this was written before the volume API was implemented. I'll update this. Thanks.